File: Vanilla/sources/zomboid/zombie/characters/BodyDamage/Metabolics.java
package zombie.characters.BodyDamage;

public enum Metabolics {
   Sleeping(0.8F),
   SeatedResting(1.0F),
   StandingAtRest(1.2F),
   SedentaryActivity(1.2F),
   Default(1.6F),
   DrivingCar(1.4F),
   LightDomestic(1.9F),
   HeavyDomestic(2.9F),
   DefaultExercise(3.0F),
   UsingTools(3.4F),
   LightWork(4.3F),
   MediumWork(5.4F),
   DiggingSpade(6.5F),
   HeavyWork(7.0F),
   ForestryAxe(8.5F),
   Walking2kmh(1.9F),
   Walking5kmh(3.1F),
   Running10kmh(6.5F),
   Running15kmh(9.5F),
   JumpFence(4.0F),
   ClimbRope(8.0F),
   Fitness(6.0F),
   FitnessHeavy(9.0F),
   MAX(10.3F);

   private final float met;

   private Metabolics(float var3) {
      this.met = var3;
   }

   public float getMet() {
      return this.met;
   }

   public float getWm2() {
      return MetToWm2(this.met);
   }

   public float getW() {
      return MetToW(this.met);
   }

   public float getBtuHr() {
      return MetToBtuHr(this.met);
   }

   public static float MetToWm2(float var0) {
      return 58.0F * var0;
   }

   public static float MetToW(float var0) {
      return MetToWm2(var0) * 1.8F;
   }

   public static float MetToBtuHr(float var0) {
      return 356.0F * var0;
   }

   // $FF: synthetic method
   private static Metabolics[] $values() {
      return new Metabolics[]{Sleeping, SeatedResting, StandingAtRest, SedentaryActivity, Default, DrivingCar, LightDomestic, HeavyDomestic, DefaultExercise, UsingTools, LightWork, MediumWork, DiggingSpade, HeavyWork, ForestryAxe, Walking2kmh, Walking5kmh, Running10kmh, Running15kmh, JumpFence, ClimbRope, Fitness, FitnessHeavy, MAX};
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/Metabolics.lua
---@class Metabolics : zombie.characters.BodyDamage.Metabolics
---@field public Sleeping Metabolics
---@field public SeatedResting Metabolics
---@field public StandingAtRest Metabolics
---@field public SedentaryActivity Metabolics
---@field public Default Metabolics
---@field public DrivingCar Metabolics
---@field public LightDomestic Metabolics
---@field public HeavyDomestic Metabolics
---@field public DefaultExercise Metabolics
---@field public UsingTools Metabolics
---@field public LightWork Metabolics
---@field public MediumWork Metabolics
---@field public DiggingSpade Metabolics
---@field public HeavyWork Metabolics
---@field public ForestryAxe Metabolics
---@field public Walking2kmh Metabolics
---@field public Walking5kmh Metabolics
---@field public Running10kmh Metabolics
---@field public Running15kmh Metabolics
---@field public JumpFence Metabolics
---@field public ClimbRope Metabolics
---@field public Fitness Metabolics
---@field public FitnessHeavy Metabolics
---@field public MAX Metabolics
---@field private met float
Metabolics = {}

---@public
---@return Metabolics[]
function Metabolics:values() end

---@public
---@return float
function Metabolics:getWm2() end

---@public
---@return float
function Metabolics:getBtuHr() end

---@public
---@return float
function Metabolics:getW() end

---@public
---@param arg0 float
---@return float
function Metabolics:MetToBtuHr(arg0) end

---@public
---@return float
function Metabolics:getMet() end

---@public
---@param arg0 float
---@return float
function Metabolics:MetToW(arg0) end

---@public
---@param arg0 float
---@return float
function Metabolics:MetToWm2(arg0) end

---@public
---@param arg0 String
---@return Metabolics
function Metabolics:valueOf(arg0) end
